packages/client-ts/src/context/engines/verida/notification/engine.ts

```diff
@@ -18,24 +19,27 @@ export default class NotificationEngineVerida implements BaseNotification {
     /**
      * Initialize notifications for the connected user
      *
-     * (ie; connect to a notification server)
+     *    * (ie; connect to a notification server)
      */
     public async init(): Promise<void> {
         this.did = (await this.context.getAccount().did()).toLowerCase()
-     }
+    }
 
-     /**
+    /**
       * Ping a notification server to fetch new messages
       */
-    public async ping(): Promise<boolean> {
-        await this.init()
-        const server = await this.getAxios()
+    public async ping(recipientContextName: string, did?: any): Promise<boolean> {
+        await this.init();
+        const server = await this.getAxios();
+       
+        this.serverUrl = this.serverUrl.endsWith("/") ? this.serverUrl : this.serverUrl + "/";
 
         try {
+            // Returns the client context and the corresponding `DID`
             await server.post(this.serverUrl + 'ping', {
                 data: {
-                    did: this.did!,
-                    context: this.context.getContextName()
+                    did: did ? did : this.did!,
```

Comment:
This is the heart of the actual bug.

The code should never have used `this.did` as that is the DID of the current user. The system should never send a notification to the current user, only to the user that the notification is being sent to.

As such, this should just be `did: did`.